src/Microsoft.Diagnostics.Monitoring.RestServer/HostRestrictionAttribute.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    /// <summary>
+    /// We want to restrict the Prometheus scraping endpoint to only the /metrics call.
+    /// To do this, we determine what port the request is on, and disallow other actions on the prometheus port.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    internal sealed class HostRestrictionAttribute : Attribute, IActionConstraintFactory
```

Comment:
nit: Rename so that its clear this is triggering off of whether metrics collection is enabled or not.

Reply:
If I can come up with a better name I will change it. The other option here is to create a configuration that represents a set of allowed ports and pair that with the attribute, rather than make the attribute mean "not prometheus ports"